### PR TITLE
Nar'sie rune can no longer be scribed off station or on shuttles

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscdel: The Nar'sie rune cannot be scribed on shuttles or off Z-level.
rscadd: The Raise Dead rune automatically grabs the ghost of the raised corpse.
/:cl:
- Made can_invoke() not require a user argument (for that auto-invoke)
- Removed unused user argument from fail_invoke, made message display an
  argument (for that auto-invoke)

### CODE REVIEW

File: code/game/gamemodes/cult/runes.dm [hunk 1]
```diff
@@ -452,6 +455,30 @@ var/list/teleport_runes = list()
...
+
+/obj/effect/rune/narsie/process()
+	// Automatically attempt a summoning, once per process
+	var/list/invokers = can_invoke()
+	if(invokers.len >= req_cultists)
+		invoke(invokers)
+	else
+		fail_invoke(message=FALSE)
+	// Automatically disappear if for some reason it's not present on station
+	if(loc.z != ZLEVEL_STATION)
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: please make sure loc is something that actually has a z(which should be true, but)

File: code/game/gamemodes/cult/runes.dm [hunk 2]
```diff
@@ -141,9 +142,11 @@ structure_check() searches for nearby cultist structures required for the invoca
...
 		animate(transform = oldtransform, alpha = 255, time = 0)
 
-/obj/effect/rune/proc/fail_invoke(var/mob/living/user)
+/obj/effect/rune/proc/fail_invoke(message=TRUE)
 	//This proc contains the effects of a rune if it is not invoked correctly, through either invalid wording or not enough cultists. By default, it's just a basic fizzle.
-	visible_message("<span class='warning'>The markings pulse with a small flash of red light, then fall dark.</span>")
+	if(message)
+		visible_message("<span class='warning'>The markings pulse with a \
+			small flash of red light, then fall dark.</span>")
 	spawn(0) //animate is a delay, we want to avoid being delayed
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: This should probably be under if(message) or it'll be spamming the 'failed invoke' animation

File: code/game/gamemodes/cult/runes.dm [hunk 3]
```diff
@@ -141,9 +142,11 @@ structure_check() searches for nearby cultist structures required for the invoca
 		animate(src, transform = matrix()*2, alpha = 0, time = 5) //fade out
 		animate(transform = oldtransform, alpha = 255, time = 0)
 
-/obj/effect/rune/proc/fail_invoke(var/mob/living/user)
+/obj/effect/rune/proc/fail_invoke(message=TRUE)
 	//This proc contains the effects of a rune if it is not invoked correctly, through either invalid wording or not enough cultists. By default, it's just a basic fizzle.
-	visible_message("<span class='warning'>The markings pulse with a small flash of red light, then fall dark.</span>")
+	if(message)
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: Do this for raise dead runes, too

File: code/game/gamemodes/cult/runes.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: also this should be ABOVE the 'attempt invoke' or people very well WILL manage to summon off-station

**Comment by ChangelingRain** [comment]
I don't really like this and would prefer it check when you try to scribe it/in can_invoke() rather than processing.
Also, this will make the rune on the cultist shuttle disappear automatically.

**Comment by Robustin** [comment]
Thank you for doing this, I've been busy.

File: code/game/gamemodes/cult/runes.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: Also I'd like to note I really liked the 'cultists trying to invoke it made it flash' thing and would be sad if it just did so automatically.
Maybe, on line 474, check if there ARE any invokers before using fail_invoke()?

**Comment by optimumtact** [comment]
Please address the comments here

**Comment by coiax** [comment]
@ChangelingRain Approve of the changes?

**Comment by Robustin** [comment]
Since there is some disagreement, I think you should consider scrapping any kind of autoinvoke. Out of the 9+ people trying to line up for summoning, you can guarantee that there are going to be multiple clicks per seconds, adding an auto-check every 2 seconds isn't going to help anything.

**Comment by ChangelingRain** [comment]
Honestly I really do not like or approve of this; the prevention on other zlevels is good, but it doesn't need the autoinvoke and it's pretty useless anyway.

**Comment by coiax** [comment]
Removing the auto invoke.

File: code/game/gamemodes/cult/runes.dm [hunk 4]
```diff
@@ -519,15 +530,15 @@ var/list/teleport_runes = list()
 	if(!potential_sacrifice_mobs.len)
 		user << "<span class='cultitalic'>There are no eligible sacrifices nearby!</span>"
 		log_game("Raise Dead rune failed - no catalyst corpses")
-		fail_invoke()
+		fail_invoke(message = FALSE)
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: No, I meant do it for the raise dead rune's fail_invoke MESSAGE

**Pull request merged**